tasks/docs.ts

```diff
@@ -7,47 +7,33 @@ const projectRootDir:string = path.resolve( __dirname, "../" );
 const sourceDir:string = path.resolve( projectRootDir, "src/" );
 const outputDir:string = path.resolve( projectRootDir, "docs/" );
 
-let environment: "development" | "production";
-
 export const docsClean:gulp.TaskFunction = () =>
 	del( [ outputDir ] );
 docsClean.displayName = "docs:clean";
 
-const setENV:( env:"development" | "production" ) => gulp.TaskFunction = env => {
+export const generateDocumentation:( env:"development" | "production" ) => gulp.TaskFunction = env => {
 	const fn:gulp.TaskFunction = () => {
-		environment = env;
-		return Promise.resolve();
+		const options:DocsEngine.Options = {
+			src: sourceDir,
+			out: outputDir,
+			mode: env,
+			logLevel: "info",
+		};
+		return DocsEngine.generate( options );
 	};
-
-	fn.displayName = "docs:set-env";
-
+	fn.displayName = "docs:generate";
 	return fn;
 };
 
-export const generateDocumentation:gulp.TaskFunction = () => {
-
-	let options:DocsEngine.Options = {
-		src: sourceDir,
-		out: outputDir,
-		mode: environment,
-		logLevel: "info",
-	};
-
-	return DocsEngine.generate(options);
-};
-generateDocumentation.displayName = "docs:build|prod";
-
 export const docsBuildDev:gulp.TaskFunction = gulp.series(
-	setENV("development"),
 	docsClean,
-	generateDocumentation
+	generateDocumentation("development")
 );
 docsBuildDev.displayName = "docs:build|dev";
 
 export const docsBuildProd:gulp.TaskFunction = gulp.series(
-	setENV("production"),
 	docsClean,
-	generateDocumentation
+	generateDocumentation("production")
 );
 docsBuildDev.displayName = "docs:build|prod";
```

Comment:
The incorrect task is been named:
```suggestion
 docsBuildProd.displayName = "docs:build|prod";
```